Init skeleton pages

rawrrencee opened the pull request:
### Changelog: 
Added components in their respective folders (Admin/Sensei/Student/Public)
Added layouts for Admin/Student/Sensei
Included dynamic menu changes in Admin/Student/Sensei layouts (MenuTop/MenuLeft custom components)
Modified existing Main layout for non-logged-in usage
Added admin, sensei, student skeleton pages (/dashboard)
Modified routers.js to include skeleton pages (navigate via url e.g. /admin/dashboard)
## Description:
This change initializes the develop branch to begin front-end development.
## Checklist:
- [x] Merged latest develop
- [x] PR title makes sense

## Screenshots (where relevant):
Student Page
![image](https://user-images.githubusercontent.com/43084055/107859840-3c7e3600-6e77-11eb-9ab1-a7221ebe5fe3.png)
Sensei Page
![image](https://user-images.githubusercontent.com/43084055/107859823-28d2cf80-6e77-11eb-9b23-507a858f3e66.png)
Admin Page
![image](https://user-images.githubusercontent.com/43084055/107859854-4dc74280-6e77-11eb-83d0-719001141e85.png)

rawrrencee committed:
initial testing

rawrrencee committed:
init sample pages, switch menu in different pages

rawrrencee committed:
add sample menu and components

natalieloke commented on src/components/Sensei/SenseiMenuBar/UserMenu/style.module.scss:
```diff
@@ -23,7 +22,7 @@
     border-bottom: 0;
     border-left: 0.3em solid transparent;
-    @include transition-slow;
+    @include transition-slow();
```
can i check if this parentheses is intended?

rawrrencee replied:
Yep, intended, I switched the menu this SenseiMenuBar was referencing from!

natalieloke commented on src/components/Sensei/SenseiMenuBar/Actions/index.js:
```diff
@@ -0,0 +1,25 @@
+
+const Actions = () => {
+  const menu = (
+    <React.Fragment>
```
we can use the new shorter syntax for react fragment which is ```<> ... </>```
https://reactjs.org/docs/fragments.html#short-syntax

rawrrencee replied:
Done!

natalieloke commented on src/components/Public/PublicMenuBar/MenuLeft/index.js:
```diff
@@ -0,0 +1,227 @@
+    const generateSubmenu = items =>
+      items.map(menuItem => {
+        if (menuItem.children) {
+          const subMenuTitle = (
+            <span key={menuItem.key}>
+              <span className={style.title}>{menuItem.title}</span>
+              {menuItem.count && <span className="badge badge-success ml-2">{menuItem.count}</span>}
+              {menuItem.icon && <span className={`${menuItem.icon} ${style.icon}`} />}
+            </span>
+          )
+          return (
+            <Menu.SubMenu title={subMenuTitle} key={menuItem.key}>
+              {generateSubmenu(menuItem.children)}
+            </Menu.SubMenu>
+          )
+        }
+        return generateItem(menuItem)
+      })
+
+    return menuData.map(menuItem => {
+      if (menuItem.roles && !menuItem.roles.includes(role)) {
+        return null
+      }
+      if (menuItem.children) {
+        const subMenuTitle = (
+          <span key={menuItem.key}>
+            <span className={style.title}>{menuItem.title}</span>
+            {menuItem.count && <span className="badge badge-success ml-2">{menuItem.count}</span>}
+            {menuItem.icon && <span className={`${menuItem.icon} ${style.icon}`} />}
+          </span>
+        )
+        return (
+          <Menu.SubMenu title={subMenuTitle} key={menuItem.key}>
+            {generateSubmenu(menuItem.children)}
+          </Menu.SubMenu>
+        )
+      }
+      return generateItem(menuItem)
+    })
+  }
```
these seem duplicated across several files in folders regarding a MenuBar. can we consider moving them to a relevant utils file where we export it and then import it to use them?

rawrrencee replied:
Somehow did not manage to export the consts used due to my limited knowledge, I don't really know how to export it so that the variables its calling are still accessible
I think need your help to try this out, @natalieloke 
Would be good to have a sample of how you would do it, as I'm still new to React, this part is quite hard for me haha

natalieloke replied:
ok will do! 👍

y1bs commented:
I agree with Nat's Point

On the admin side, every else looks good! Thanks Lawrence!

rawrrencee committed:
Modify starter files according to PR suggested changes

TODO: export common functions to utils folder for usage across multiple common components

rawrrencee committed:
responsive fix for custom menuTop, switch menu for student in browse and student in dashboard, add new menus and pages

fixed top menu bar for iPad view by hiding search bar

TODO: extract common components, clean up components folder

rawrrencee committed:
Add base landing page, Add account redirect upon logging in, Modify Layout names according to convention, Refactor some page names from CleanUI, Further component changes

Reminder to self: Clean up customised components
